heidrun/getty.rb

```diff
@@ -0,0 +1,127 @@
+DATA_PROVIDER_MAP = {
+  'GETTY_ROSETTA' => 'Getty Research Institute',
+  'GETTY_OCP' => 'Getty Research Institute'
+}
+
+build_url = lambda do |p|
+  source = p.root[Krikri::PrimoParser.record('control', 'sourceid').join('/')]
+           .first.value
+  id = p.root[Krikri::PrimoParser.record('control', 'recordid').join('/')]
+       .first.value
+
+  base_url = 'http://primo.getty.edu/primo_library/libweb/action/dlDisplay.do'
+
+  case source
+  when 'GETTY_ROSETTA'
+    "#{base_url}?vid=GRI&afterPDS=true&institution=01GRI&docId=#{id}"
+  when 'GETTY_OCP'
+    "#{base_url}?vid=GRI-OCP&afterPDS=true&institution=01GRI&docId=#{id}"
+  end
+end
+
+lookup_language = lambda do |i|
+  val = i.value.downcase
+  (val == 'multiple languages') ? 'mul' : val
+end
+
+Krikri::Mapper.define(:getty,
+                      parser: Krikri::PrimoParser) do
+  provider class: DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/getty'
+    label 'J. Paul Getty Trust'
+  end
+
+  dataProvider class: DPLA::MAP::Agent,
+               each: record.field(*Krikri::PrimoParser.record('control',
+                                                              'sourceid'))
+                      .map { |i| DATA_PROVIDER_MAP.fetch(i.value, i.value) }
+                      .first_value,
+               as: :provider do
+    label provider
+  end
+
+  isShownAt class: DPLA::MAP::WebResource do
+    uri build_url
+  end
+
+  preview class: DPLA::MAP::WebResource do
+    uri record.field('sear:LINKS', 'sear:thumbnail')
+  end
+
+  sourceResource class: DPLA::MAP::SourceResource do
+    # TODO: Enrichment needed to split title fields on semicolons
```

Comment:
Should be set, here: https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/split_at_delimiter.rb